src/imperative/imperative_utils.h

```diff
@@ -963,13 +963,13 @@ inline void CreateEngineOpSeg(
     seg_execs.push_back(exec);
 
     auto& seg = (*opr_segs)[nid];
-    if (is_async) {
-      seg = EngineOprSeg{false, nid + 1};
-      seg.opr.reset(CreateEngineOp(default_ctx, seg_execs));
+    if (!valid) {
+      seg = EngineOprSeg{false, nid + 1, nullptr};
       seg_execs.clear();
       seg_start = nid + 1;
-    } else if (!valid) {
-      seg = EngineOprSeg{false, nid + 1, nullptr};
+    } else if (is_async) {
+      seg = EngineOprSeg{false, nid + 1};
+      seg.opr.reset(CreateEngineOp(default_ctx, seg_execs));
       seg_execs.clear();
       seg_start = nid + 1;
```

Comment:
extract common code outside branch

Reply:
we can't. the code is used in "if" and "else if". it's not used in "else"